woocommerce/woocommerce
# With Product Variations HOC > Convert to TypeScript

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR converts the Product Variations HOC from JavaScript to TypeScript.

Closes #42361 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Login to the dashboard.
2. Create a new post (Posts > Add new post)
3. In the editor, you should see a block inserter icon or an option to "Add Block". Click on it.
4. In the block inserter, search for "Reviews" or browse through the available blocks until you find it.
    ![CleanShot 2024-02-14 at 17 37 01@2x](https://github.com/woocommerce/woocommerce/assets/20469356/f40ddfdf-cc93-4a6f-ba51-8a1a65c6c76c)
6. Click on the "Reviews by Product" blocks to add it to your content.
7. Using the Product selector, select a product **without** reviews.
8. Make sure everything is working correctly.
9. Select a product **with** reviews using the Product selector.
10. Make sure everything is working correctly.
11. Save the post and visit it on the frontend to ensure they are working as expected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Convert With Product Variations HOC to TypeScript

</details>
